api/src/main/java/jakarta/config/Config.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package jakarta.config;
+
+import java.util.Optional;
+
+import jakarta.config.spi.ConfigSource;
+import jakarta.config.spi.Converter;
+
+/**
+ * Access to configuration values.
+ */
+public interface Config {
+    /**
+     * Return all the currently registered {@link jakarta.config.spi.ConfigSource sources} for this configuration.
+     * <p>
+     * The returned sources will be sorted by priority and name, which can be iterated in a thread-safe
+     * manner. The {@link java.lang.Iterable Iterable} contains a fixed number of {@link jakarta.config.spi.ConfigSource
+     * configuration
+     * sources}, determined at application start time, and the config sources themselves may be static or dynamic.
+     *
+     * @return the configuration sources
+     */
+    Iterable<ConfigSource> getConfigSources();
```

Comment:
To me ...
Iterable and List implies ordered sources.
Stream doesn't.

Why can't this just be a new class `ConfigSources`? which abstracts the whole idea, provides multiple ways to access the list, etc.

Reply:
It can. One discussed possibility was to split sources and converters into their own separate APIs, since everything ends up in `Config`. Then `Config` could become the actual configuration value (or values) representation.

Reply:
I think we discussed this in the past for performance reasons to use Iterable. Also it is an ordered list.